apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/sink/direct/FissClaimRdaSink.java

```diff
@@ -13,10 +13,19 @@
 import gov.cms.mpsm.rda.v1.FissClaimChange;
 import java.io.IOException;
 import java.util.List;
+import java.util.regex.Pattern;
 import javax.annotation.Nonnull;
 
 /** Implementation of AbstractClaimRdaSink that adds FISS claim specific methods. */
 public class FissClaimRdaSink extends AbstractClaimRdaSink<FissClaimChange, RdaFissClaim> {
+  /**
+   * Some FISS claim records received from the IDR are not real claims. These records can be
+   * recognized by their DCN values. Any FISS claim with DCN that is all zeros or ends with XXX is
+   * not a real claim and should not be written to the database. This regex is used by {@link
+   * #isValidMessage} to recognize these bad claims.
+   */
+  private static final Pattern InvalidDcnRegex = Pattern.compile("(^0+$)|(XXX$)");
```

Comment:
We need to account for leading and trailing whitespace as well.  For example these are two DCNs from PROD that we'd want to skip over.
- '22239999999999XXX      ' 
- ' 00000000000           '

Reply:
Also when reviewing the database, I just noticed something new.  Its not just that the DCN needs to end in 'XXX', its that the [BSI portion](https://confluence.cms.gov/pages/viewpage.action?pageId=540083018) (positions 15-17) of the DCN is 'XXX'.  For example

- '22230501299999XXX4     '

Sorry for the new spec, I just noticed this myself.